### ISSUE TITLE
shortnames are not unique

### ISSUE THREAD

**Issue by adcxyz** [ModalityTeam/Modality-toolkit#210]
currently the shortname algorithm creates identical shortnames for 
runNdrive + rnnd wireless, and the three Steinberg CMC devices. 
this makes the descriptions indistinguishable when doing MKtl(\short0); 

this touches issues #188, and  #168, #209 (and maybe #185)

Possible solutions I see: (#188)
1. keep the shortname as is, 
    and if there are multiple matches, check by actual idInfo reported
    -> should work works for HID and MIDI, not for virtual devices or OSC
2. for non-unique shortnames, 
    require filename (i.e. eturn nil and post request for MKtl(name, filename), 
    and then prefer filename for device ID
    -> would be systematic and clear, maybe a bit cumbersome
3. change the shortname algorithm so the ones for known devices become unique
    -> would work now, new devices might clash again in the future
4. use full idInfo in shortname 
    -> ugly, and as in #209 still not unique for MIDI devices which register multiply  
    under the same name and Port 1, Port 2, Port 3 (e.g. the QuNexus)

I would propose that 
option 2 should be made to work, (unique filename should win over nonunique shortname), 
and 3, shortname algorithm should create unique names.
[and to solve #209, use solution proposed there]

**Comment by LFSaw** [comment]
My 2c:
Get rid of the shortnames, I say.

How about:

```
MKtl(\ctl, "nanoKontrol");
```

searches for all available devices connected to the system that match with the identifier string and 

+ if only one found: opens that one
+ if more than one found: attaches first unopened device and tells the user that there are more to be connected.

```
MKtl(\ctl, "nanoKontrol", idx: 5);
```

+ attaches 6th device that matches.

**Comment by bhabbestad** [comment]
+1 for getting rid of #shrtnms
good for clarity

**Comment by adcxyz** [comment]
OK, closed with 4b3612ca03544b544adf1fcc4babff5d08004959 
lookupNames are really unique now, and pretty clear to read, as here:
( 'midi': List[ \midiSrc_0_nanokontrol, 
\midiDest_0_nanokontrol ], 
'hid': List[ \hid_6_run_n_drive ] )

**Comment by LFSaw** [comment]
I like the lookupNames. Thanks for this!


:)
	Till
